### PR TITLE
Deploy test, preprod and production environments for dra-dqt-notifications

### PR DESCRIPTION
**Pull request by enformatique11**
Deployment of dra-dqt-notification to test, preprod and production environments.
Added additional app service plan instance for preprod and production.
All the environment implementations are successful.

Dev: https://portal.azure.com/#@platform.education.gov.uk/resource/subscriptions/9816b7b0-58ca-4972-b25b-c7dbd3ee1c6d/resourceGroups/s165d01-dqtnoti-dv-rg/overview
Test: https://portal.azure.com/#@platform.education.gov.uk/resource/subscriptions/250ff557-239a-4d89-83a4-07bdfc6218c7/resourceGroups/s165t01-dqtnoti-ts-rg/overview
Preprod: https://portal.azure.com/#@platform.education.gov.uk/resource/subscriptions/250ff557-239a-4d89-83a4-07bdfc6218c7/resourceGroups/s165t01-dqtnoti-pp-rg/overview
Production: https://portal.azure.com/#@platform.education.gov.uk/resource/subscriptions/9c2dcb72-4bba-4acc-8d5f-8b8389a68238/resourceGroups/s165p01-dqtnoti-pd-rg/overview

### CODE REVIEW

File: terraform/variables.tf [hunk 1]
```diff
@@ -30,10 +30,19 @@ variable "azure_sp_credentials_json" {
...
+  type    = bool
+  default = false
+}
+
 variable "resource_group_name" {
   type = string
 }
 
+variable "notify_storage_account_name" {
+  default = null
```
Comment (NickGraham101) [inline terraform/variables.tf:43]: This isn't an optional variable so don't set a default

File: terraform/workspace_variables/dev.tfvars.json [hunk 2]
```diff
@@ -2,5 +2,7 @@
   "environment_name": "dev",
   "key_vault_name": "s165d01-dqtnoti-dv-kv",
   "resource_group_name": "s165d01-dqtnoti-dv-rg",
-  "resource_prefix": "s165d01"
+  "resource_prefix": "s165d01",
+  "enable_blue_green": false,
+  "notify_storage_account_name": "s165d01dqtnotinotifydv"
```
Comment (NickGraham101) [inline terraform/workspace_variables/dev.tfvars.json:7]: This name could be improved, calling it having `noti` in the name twice isn't helpful.  It should either be `s165d01dqtnotidv` or `s165d01dqtnotiappdv` (app since at the minute the storage is just being used to back the function app)

File: terraform/variables.tf (hunk 1 above)
Comment (enformatique11) [reply terraform/variables.tf]: fixed

File: .github/workflows/build-and-deploy-notifications.yml [hunk 3]
```diff
@@ -64,6 +64,12 @@ jobs:
           report_name: "Test results"
           dotnet_test_args: '-e ConnectionStrings__DefaultConnection="Data Source=(local); Initial Catalog=${{ env.SQL_DB }}; User=sa; Password=${{ env.SQL_PASSWORD }}; TrustServerCertificate=True"'
 
+      - name: Upload artifact for deployment job
```
Comment (NickGraham101) [inline .github/workflows/build-and-deploy-notifications.yml:67]: No steps have been added to consume this artifact so it might be better off in a different PR.  It will also probably need to be more selective in what's uploaded for deployment to the function app service

File: terraform/workspace_variables/dev.tfvars.json (hunk 2 above)
Comment (enformatique11) [reply terraform/workspace_variables/dev.tfvars.json]: fixed

File: .github/workflows/build-and-deploy-notifications.yml (hunk 3 above)
Comment (enformatique11) [reply .github/workflows/build-and-deploy-notifications.yml:67]: Okay make sense

Comment (enformatique11) [reply .github/workflows/build-and-deploy-notifications.yml:67]: removed

**Review by NickGraham101 — APPROVED** [review]

**Comment by enformatique11** [comment]
[tra-dqt-notifications complete deployment workflow](https://trello.com/c/8NFgJzVF/450-tra-dqt-notifications-complete-deployment-workflow)
